### DISCUSSION TITLE
Localization support?

### DISCUSSION THREAD

**Discussion by eh-jogos** [dialogic-godot/dialogic#916] [Q&A] [answered]
Hi! 
Haven't actually used Dialogic in a project yet, but was trying to glance at it's documentation to get an idea of how it works to see if it could replace a dialogue system in another project, if it's a good fit and all.

One thing I couldn't find at a glance and would be needed for the project is localization support. How do you deal with localization in dialogic?

Should I just replace the actual text from dialog events for Localization String ids? or is there another way to do it that escaped me?

**Comment by zaknafean** [comment, marked as answer]
I'm afraid at this time we don't really support it at this time.
We have a few PRs that handle it in different ways, but we haven't settled on a method of how it is going to be handled yet. The decision on that probably won't happen until Dialogic 2.0

**Reply by eh-jogos** [reply]
Thanks!

**Reply by Jowan-Spooner** [reply]
Small update for people looking at this later: Dialogic 2.0 (currently WIP) now has a translation system that adds unique id's to translatable events and can automatically create/update csv files. You can have one big csv file or one for each timeline.

**Reply by eddex** [reply]
It would be really nice if it would work with Godot's built in localization system and the POT generation feature. Is this what will be implemented in Dialogic 2.0 or will we have to maintain the Dialogic strings separately from the UI translations?

EDIT: I had a look for myself. Godot generates a `.pot` file for translations of e.g. UI elements. These can be used with a tool like `poedit` to generate translations in `.po` files that can then be imported in `Project Settings > Localization > Translations`.
Dialogic 2 on the other hand generates `.csv` files, that can be translated using any CSV editor. After editing you can press a button in Dialogic to automatically add the translations to Godot's `Project Settings > Localization > Translations` :tada:.
While translating needs to be done differently for the two, both can then be controlled using Godot's built-in `TranslationServer`. It's very convenient!
